src/core/mac/mac.cpp

```diff
@@ -958,6 +972,8 @@ extern "C" void otPlatRadioTxDone(otInstance *aInstance, otRadioFrame *aFrame, o
 {
     otLogFuncEntryMsg("%!otError!", aError);
 
+    aInstance->mThreadNetif.GetMac().SetFramePending((static_cast<Frame *>(aAckFrame))->GetFramePending());
```

Comment:
Also here I think we need to check and remember the "pending" on the sent frame `aFrame` instead of the `aAckFrame`.